### PR TITLE
Ports eyes adjusting to the dark based on species darksight

### PR DESCRIPTION
**Pull request by Aurorablade**
Ports https://github.com/PolarisSS13/Polaris/pull/4626

I need tiger to help look at the code cuase he brought it to my attention..and how diffrent the stuff is..

...and i can't tell if darksight is affecting me or not to test sometimes ANYWAY...(i need a better monitor)

:cl: Fethas
tweak: eyes will now adjust to darkness based on the species darksight level
/:cl:

### CODE REVIEW

**Comment by Tayyyyyyy** [comment]
This is really cool. It's a buff to darksight, but it makes much more sense to have this system than make you use mesons for darksight. Screenshots of what it looks like to have maximum darksight?

**Comment by Aurorablade** [comment]
I am at work so you may need to look at what i am porting from to see the shot therr

**Comment by tigercat2000** [comment]
@Tayyyyyyy You realize that the "harsh black" darkness hasn't been a thing for months, right? You can see fine in the dark without mesons as it is. Literally all this PR does is introduce a fade-in/out effect, it literally is just a big black square layered on the screen behind the lighting layer.

**Comment by Aurorablade** [comment]
Yeah I kinda made this out the door to work so I need to retitle this and close my other pr as this isn't really a balance thing. But again at work

**Comment by KasparoVy** [comment]
God bless.

**Comment by Aurorablade** [comment]
yeah still unsure if the overlay is working so need a extra code check up.

**Comment by KasparoVy** [comment]
I just tested this and it doesn't seem to work. The overlay straight-up doesn't appear at all, regardless of area brightness or the amount of dark_view my eyes have..

**Comment by KasparoVy** [comment]
I just tested this and it doesn't seem to work. The overlay straight-up doesn't appear at all, regardless of area brightness or the amount of dark_view my eyes have.

I assume it's because I can't see any code incorporating this with our HUD system or wherever this overlay should end up in our mob's vars.

@Aurorablade This PR is missing all the other parts of PolarisSS13/Polaris#4626 that tie this into HUD handling. See [file 1](https://github.com/PolarisSS13/Polaris/pull/4626/files#diff-156a17a44f59f1c997481143d186cd2e), [file 2](https://github.com/PolarisSS13/Polaris/pull/4626/files#diff-56766bbbacdf1f25619eef735b1bdad8), and maybe these files (no idea if our lighting system is even remotely similar): [file 3](https://github.com/VOREStation/Polaris/blob/a864e4176b41fceb687f6eccde105317084107db/code/modules/lighting/lighting_overlay.dm) & [file 4](https://github.com/VOREStation/Polaris/blob/a864e4176b41fceb687f6eccde105317084107db/code/__defines/mobs.dm).

**Comment by Aurorablade** [comment]
it not similliar meaning i need to think of a diffrent way to do things cleanly.

**Comment by KasparoVy** [comment]
I've got it sort-of working @Aurorablade , though the darkview overlay doesn't have the desired effect due to our lack of plane-master business.
This is because the plane handling allows for the isolated blending of effects ONTO lighting BEFORE lighting is blended onto everything else, which means that instead of brightening the final result of a mush of other overlays it just makes the 'lighting' more pronounced.
Thus having the effect of making an area appear as though it's more well lit.

_See below and compare with the .gif in the Polaris port. **This needs planes to function properly.**_
darksight_adjusting_busted

### ---------------------------------------------------------

Looking over how VORE & Polaris have done it, it's utterly ascended business that I just can't wrap my head around. So many questions, not least of which are 'how the heck did they figure out the matrices for the different kinds of colourblindness'?

Mushed my mind. I've no idea how they developed that.
For clues, see [here](https://github.com/PolarisSS13/Polaris/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Amerged+plane).

Anyway, [here's my commit with the progress](https://github.com/KasparoVy/Paradise/commit/289336834b260808695ad9290b0dbced0847e3ee). I don't see how we can do this without porting their Plane system & changing mob icon update code along with it.

### **tl;dr**: _This is not how it's supposed to work, and this feature should come after a port of planes and mob icon generation changes._

**Comment by Aurorablade** [comment]
well the goal now is to figure out how to use it without the plane stuff...
WE DO need mob icon code updates thought...badly...

**Comment by Aurorablade** [comment]
now one thought is since it useing a DMI for the overlay can we somehow adjust the DMI or make that would suit the purpose.

**Comment by KasparoVy** [comment]
I'm not going to dare try to meme this into working w/o the planes, they go hand-in-hand. This voodoo is way above me.

**Comment by Aurorablade** [comment]
so yeah this needs planes. and i got alot of stress irl right now so i dunno if i can wrap my head around porting that and from what codebase (though tg would be good maybe?) if anyone wants to port some plane stuff please do (my birthday is in may..get a head start and i will hug you)
Gonna close for now until someone gives me a good work around.
